### ISSUE TITLE
Support for SNI-based backends with a wildcard cert

### ISSUE THREAD

**Issue by tstm** [varnish/hitch#86]
In many organizations wildcard certs are in use. There's one frontend but many backends and traffic is redirected using the SNI information. ie:

Frontend:
*.example.com

Backends:
one.example.com
two.example.com
default.example.com

This configuration is currently not possible using hitch. At least in our use case this would be very useful. It probably would not require much change?

**Comment by AxisNL** [comment]
We've got about 20 wildcard certs on our balancers, this works fine. Hitch passes all decrypted traffic to varnish, and varnish will then decide on which backend to forward to.

**Comment by lkarsten** [comment]
Hi.

I've added this to the list of possible future features in the wiki: https://github.com/varnish/hitch/wiki/Future-features

On a personal note, I'm not sure we want this as it would make the backend selection logic more complicated. Doing the selection in Varnish as suggested is a good workaround. If you feel strongly about it and want to make a stab at implementing it, let us know and we'll help you along.

Closing this issue as we only keep actionable bugs in the issue tracker.
